python/lsst/meas/extensions/multiprofit/rebuild_coadd_multiband.py

```diff
@@ -0,0 +1,248 @@
+                    quantum = node.quantum
+                    break
+            if quantum is None:
+                raise ValueError(
+                    f"{dataId=} not found in {[x.quantum.dataId for x in quantumgraph.outputQuanta]=}"
+                )
+        taskDef = next(iter(quantumgraph.iterTaskGraph()))
+        butlerQC = QuantumContext(butler, quantum)
+        config = butler.get(f"{taskDef.label}_config")
+        # I have no idea what to put for initInputs.
+        # quantum.initInputs looks wrong - the values can be lists
+        # quantumgraph.initInputRefs(taskDef) returns a list of DatasetRefs...
+        # ... but I'm not sure how to map that to connection names?
+        task: CoaddMultibandFitTask = taskDef.taskClass(config=config, initInputs={})
+        if not isinstance(task, CoaddMultibandFitTask):
+            raise ValueError(f"{task=} type={type(task)} !isinstance of {CoaddMultibandFitTask=}")
+        task_fit: MultiProFitSourceTask = task.fit_coadd_multiband
+        if not isinstance(task_fit, MultiProFitSourceTask):
+            raise ValueError(f"{task_fit=} type={type(task_fit)} !isinstance of {MultiProFitSourceTask=}")
+        inputRefs, outputRefs = taskDef.connections.buildDatasetRefs(quantum)
+        inputs = butlerQC.get(inputRefs)
+        catexps = task.build_catexps(butlerQC, inputRefs, inputs)
+        catexps = [task_fit.make_CatalogExposurePsfs(catexp) for catexp in catexps]
+        cat_output: astropy.table.Table = butler.get(outputRefs.cat_output, storageClass="ArrowAstropy")
+        return cls(
+            catexps=catexps, task_fit=task_fit, catalog_multi=inputs["cat_ref"], fit_results=cat_output,
+        )
+
+    def make_config_data(self):
+        config_data = CatalogSourceFitterConfigData(channels=self.channels, config=self.task_fit.config)
+        return config_data
+
+    def make_model(
+        self, idx_row: int, config_data: CatalogSourceFitterConfigData = None, init: bool = True,
+    ) -> g2f.Model:
+        if config_data is None:
+            config_data = self.config_data
+        model = self.task_fit.get_model(
+            idx_row=idx_row,
+            catalog_multi=self.catalog_multi,
+            catexps=self.catexps,
+            config_data=config_data,
+            results=self.fit_results,
+            set_flux_limits=False,
+        )
+        if init:
+            self.set_model(idx_row, config_data)
+        return model
+
+    def set_model(self, idx_row: int, config_data: CatalogSourceFitterConfigData = None) -> None:
+        if config_data is None:
+            config_data = self.config_data
+        row = self.fit_results[idx_row]
+        prefix = config_data.config.prefix_column
+        offsets = {}
+        offset_cen = config_data.config.centroid_pixel_offset
+        if offset_cen != 0:
+            offsets[g2f.CentroidXParameterD] = -offset_cen
+            offsets[g2f.CentroidYParameterD] = -offset_cen
```

Comment:
So the models are always square? Is there any chance that will change in the future?

Reply:
Ah, no, these aren't bounding box offset but coordinate system offsets. MultiProFit defines the bottom-left corner of the bottom-left pixel as (0,0)* whereas it's (-0.5, -0.5) in the stack. I left that parameter general but I found it even less likely that someone would need to make it asymmetric than to give a value other than 0 or 0.5.

*I might regret this choice in the future but it's a bit late to change now.